libs/payments/cart/src/lib/cart.utils.ts

```diff
@@ -26,3 +27,35 @@ export function stripeErrorToErrorReasonId(
       return CartErrorReasonId.Unknown;
   }
 }
+
+export function handleEligibilityStatus(eligibility: EligibilityStatus) {
+  const eligibilityStatus = handleEligibilityStatusMap[eligibility];
+
+  switch (eligibility) {
+    case 'create':
+    case 'upgrade':
+      return {
+        eligibilityStatus: eligibilityStatus,
+        state: CartState.START,
+        errorReasonId: null,
+      };
+    case 'downgrade':
+      return {
+        eligibilityStatus: eligibilityStatus,
+        state: CartState.FAIL,
+        errorReasonId: CartErrorReasonId.BASIC_ERROR,
+      };
+    case 'blocked_iap':
+      return {
+        eligibilityStatus: eligibilityStatus,
+        state: CartState.FAIL,
+        errorReasonId: CartErrorReasonId.IAP_UPGRADE_CONTACT_SUPPORT,
+      };
+    default:
+      return {
+        eligibilityStatus: eligibilityStatus,
+        state: CartState.FAIL,
+        errorReasonId: CartErrorReasonId.Unknown,
+      };
+  }
```

Comment:
Since this is a 1:1 lookup, I'd recommend we use a map rather than a switch.

Reply:
Will update!